shared/src/commonMain/kotlin/org/hyperskill/app/track/data/repository/TrackRepositoryImpl.kt

```diff
@@ -1,15 +1,33 @@
 package org.hyperskill.app.track.data.repository
 
+import org.hyperskill.app.core.domain.repository_cache.RepositoryCacheProxy
+import org.hyperskill.app.track.data.source.TrackCacheDataSource
 import org.hyperskill.app.track.data.source.TrackRemoteDataSource
 import org.hyperskill.app.track.domain.model.Track
 import org.hyperskill.app.track.domain.repository.TrackRepository
 
 class TrackRepositoryImpl(
-    private val trackRemoteDataSource: TrackRemoteDataSource
+    private val trackRemoteDataSource: TrackRemoteDataSource,
+    private val trackCacheDataSource: TrackCacheDataSource
 ) : TrackRepository {
-    override suspend fun getTracks(trackIds: List<Long>): Result<List<Track>> =
-        trackRemoteDataSource.getTracks(trackIds)
+
+    private val trackCacheProxy = RepositoryCacheProxy(
+        cache = trackCacheDataSource,
+        loadValuesFromRemote = { trackIds ->
+            trackRemoteDataSource.getTracks(trackIds)
+        },
+        getKeyFromValue = { track ->
+            track.id
+        }
```

Comment:
The alternative is to replace `track` with `it` so that you don't need to name the lambda parameter.
I decided to name parameters in this case for better readability.